spec/requests/api/orchestration_template_spec.rb

```diff
@@ -125,4 +125,83 @@
       expect(OrchestrationTemplate.exists?(hot.id)).to be_falsey
     end
   end
+
+  context 'orchestration template copy' do
+    before do
+      @orchestration_template = FactoryGirl.create(:orchestration_template_cfn)
+      @new_content            = "{ 'Description': 'Test content 1' }\n"
+    end
+
+    it 'forbids orchestration template copy without an appropriate role' do
+      api_basic_authorize
+
+      run_post(
+        orchestration_templates_url(@orchestration_template.id),
+        gen_request(:copy, :content => @new_content)
+      )
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'forbids orchestration template copy with no content specified' do
+      api_basic_authorize collection_action_identifier(:orchestration_templates, :copy)
+
+      run_post(orchestration_templates_url(@orchestration_template.id), gen_request(:copy))
+
+      expect(response).to have_http_status(:bad_request)
+    end
+
+    it 'can copy single orchestration template with a different content' do
+      api_basic_authorize collection_action_identifier(:orchestration_templates, :copy)
+
+      expected = {
+        'content'     => @new_content,
+        'name'        => @orchestration_template.name,
+        'description' => @orchestration_template.description,
+        'draft'       => @orchestration_template.draft,
+        'orderable'   => @orchestration_template.orderable
+      }
+
+      expect do
+        run_post(
+          orchestration_templates_url(@orchestration_template.id),
+          gen_request(:copy, :content => @new_content)
+        )
+      end.to change(OrchestrationTemplateCfn, :count).by(1)
+
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+      expect(response.parsed_body['id']).to_not equal(@orchestration_template.id)
+    end
+
+    it 'can copy multiple orchestration templates with a different content' do
+      api_basic_authorize collection_action_identifier(:orchestration_templates, :copy)
+
+      orchestration_template_2 = FactoryGirl.create(:orchestration_template_cfn)
+      new_content_2 = "{ 'Description': 'Test content 2' }\n"
```

Comment:
I think it because we add new line in the orchestration template by intention(I can see it from code)
text.encode(:universal_newline => true).chomp.concat("\n")